### DISCUSSION TITLE
DBAL SQL query and DBAL Custom types

### DISCUSSION THREAD

**Discussion by berkut1** [symfony/symfony#48018] [Q&A]
Hello,

If I use custom doctrine types https://www.doctrine-project.org/projects/doctrine-dbal/en/current/reference/types.html
and SQL query builder https://www.doctrine-project.org/projects/doctrine-dbal/en/current/reference/query-builder.html#sql-query-builder
For example:
```
$this->connection->createQueryBuilder()
            ->select(
                'id', //where id should be a custom doctrine type
                'login',
            )
            ->from('user_users')
            ->executeQuery()->fetchAllAssociative();
```

Isn't there a way that doctrine automatically converts it to types?
Do I understand correctly that custom doctrine types only work with the Entity Query Builder?

The documentation also confuse me https://symfony.com/doc/current/doctrine/dbal.html

Thanks.

**Comment by berkut1** [comment]
Okay, there is a solution https://www.doctrine-project.org/projects/doctrine-orm/en/2.13/reference/native-sql.html
And from `createQueryBuilder()` get `getSQL()`
P.S But unfortunately doesn't work with DTOs. :)

**Reply by stof** [reply]
Doctrine DBAL does not perform any automatic conversion of query results, because there is no hydration step for that (this is also true for core types like the date types for instance).
When using the ORM, you can configure its hydration step to convert the result with your custom type.
